src/istore.c

```diff
@@ -976,3 +976,73 @@ istore_svals(PG_FUNCTION_ARGS)
 
     SRF_RETURN_DONE(funcctx);
 }
+
+/*
+ * A boolean function that supports `select '1=>1'::istore = '2=>2'::istore`
+ */
+PG_FUNCTION_INFO_V1(istore_eq);
+Datum
+istore_eq(PG_FUNCTION_ARGS)
+{
+    IStore *is1, *is2;
+    IStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
```

Comment:
oh no the equality check for `NULL` is well defined in SQL 
> Ordinary comparison operators yield null (signifying "unknown"), not true or false, when either input is null.

imagine you run a `WHERE column = '1=>1'::istore` we'd definitely not want an error if `column` is NULL